engine/gameobject/src/dmsdk/gameobject/component.h

```diff
@@ -666,24 +683,27 @@ namespace dmGameObject
         // Workaround for dead-stripping on OSX/iOS. The symbol "name" is explicitly exported. See wscript "exported_symbols"
         // Otherwise it's dead-stripped even though -no_dead_strip_inits_and_terms is passed to the linker
         // The bug only happens when the symbol is in a static library though
-        #define DM_REGISTER_COMPONENT_TYPE(symbol, desc, name, type_create_fn) extern "C" void __attribute__((constructor)) symbol () { \
-            dmGameObject::RegisterComponentTypeDescriptor((struct dmGameObject::ComponentTypeDescriptor*) &desc, name, type_create_fn); \
+        #define DM_REGISTER_COMPONENT_TYPE(symbol, desc, name, type_create_fn, type_destroy_fn) extern "C" void __attribute__((constructor)) symbol () { \
+            dmGameObject::RegisterComponentTypeDescriptor((struct dmGameObject::ComponentTypeDescriptor*) &desc, name, type_create_fn, type_destroy_fn); \
         }
     #else
-        #define DM_REGISTER_COMPONENT_TYPE(symbol, desc, name, type_create_fn) extern "C" void symbol () { \
-            dmGameObject::RegisterComponentTypeDescriptor((struct dmGameObject::ComponentTypeDescriptor*) &desc, name, type_create_fn); \
+        #define DM_REGISTER_COMPONENT_TYPE(symbol, desc, name, type_create_fn, type_destroy_fn) extern "C" void symbol () { \
+            dmGameObject::RegisterComponentTypeDescriptor((struct dmGameObject::ComponentTypeDescriptor*) &desc, name, type_create_fn, type_destroy_fn); \
             }\
             int symbol ## Wrapper(void) { symbol(); return 0; } \
             __pragma(section(".CRT$XCU",read)) \
             __declspec(allocate(".CRT$XCU")) int (* _Fp ## symbol)(void) = symbol ## Wrapper;
     #endif
 
-
-    /*#
+    /*# register a new component type
+     * @param symbol [type: cpp_symbol_name] The unique C++ symbol name
+     * @param name [type: const char*] name of the component type (i.e. the resource suffix)
+     * @param create_fn [type: dmGameObject::Result (*fn)(const ComponentTypeCreateCtx* ctx, ComponentType* type)] The type configuration function. May not be 0.
+     * @param destroy_fn [type: dmGameObject::Result (*fn)(const ComponentTypeCreateCtx* ctx, ComponentType* type)] The type destruction function. May be 0.
      */
-    #define DM_DECLARE_COMPONENT_TYPE(symbol, name, type_create_fn) \
+    #define DM_DECLARE_COMPONENT_TYPE(symbol, name, type_create_fn, type_destroy_fn) \
```

Comment:
A breaking change!

Only used in 2 extensions (spine+rive), so should create new fix-branches for those.